uxarray/grid.py

```diff
@@ -540,6 +542,73 @@ def _build_edge_node_connectivity(self, repopulate=False):
         setattr(self, "Mesh2_edge_nodes", self.ds['Mesh2_edge_nodes'])
         setattr(self, "nMesh2_edge", edge_nodes_unique.shape[0])
 
+    def _build_node_faces_connectivity(self):
+        """Builds the `Grid.Mesh2_node_faces`: integer DataArray of size
+        (nMesh2_node, MaxNumFacesPerNode) (optional) A DataArray of indices
+        indicating faces that are neighboring each node.
+
+        This function converts the face-node connectivity data into a sparse matrix, and then constructs the node-face
+        connectivity by iterating over each node in the mesh and retrieving the set of neighboring faces.
+
+        Raises
+        ------
+        RuntimeError
+            If the Mesh object does not contain a 'Mesh2_face_nodes' variable.
+        """
+        if "Mesh2_face_nodes" not in self.ds:
+            raise RuntimeError(
+                "Currently, node_face_connectivity can only be built with Mesh2_face_nodes"
+            )
+
+        # First we need to build a matrix such that: the row indices are face indexes and the column indices are node
+        # indexes (similar to an adjacency matrix)
+        face_indices, node_indices, non_zero_element_flags = _convert_face_node_conn_to_sparse_matrix(
+            self.ds["Mesh2_face_nodes"].values)
+        coo_matrix = sparse.coo_matrix(
+            (non_zero_element_flags, (node_indices, face_indices)))
+        csr_matrix = coo_matrix.tocsr()
+        # get the row and column indices of the non-zero elements
+        rows, cols = csr_matrix.nonzero()
+        # Find the frequency of each face to determine the maximum number of faces per node
+        freq = np.bincount(rows)
+        nMaxNumFacesPerNode = freq.max()
+
+        node_face_connectivity = [[]] * self.nMesh2_node
+
+        # find the indices where the array changes value
+        change_indices = np.where(np.diff(rows) != 0)[0] + 1
+
+        # split the array at the change indices to get subarrays of consecutive same elements
+        subarrays = np.split(rows, change_indices)
+
+        # get the start and end indices for each subarray
+        start_indices = np.cumsum(
+            [0] + [len(subarray) for subarray in subarrays[:-1]])
+        end_indices = np.cumsum([len(subarray) for subarray in subarrays]) - 1
+
+        for node_index in range(self.nMesh2_node):
+            node_face_connectivity[node_index] = cols[
+                start_indices[node_index]:end_indices[node_index] + 1]
+            if len(node_face_connectivity[node_index]) < nMaxNumFacesPerNode:
+                node_face_connectivity[node_index] = np.append(
+                    node_face_connectivity[node_index],
+                    np.full(
+                        nMaxNumFacesPerNode -
+                        len(node_face_connectivity[node_index]),
+                        self.ds["Mesh2_face_nodes"].attrs["_FillValue"]))
```

Comment:
When referencing the fill value, use the important constant `INT_FILL_VALUE` instead.

Reply:
This is the point that has bothering me for a while:

1. If we allow the user to customize the `fill_value`, why do we keep removing the customer fill_value support from our functions? 
2. If we don't allow the user to customize the `fill_value`, why do we store the "_FillValue" `attrs` in the Grid.ds`?

Reply:
1. The user currently can't customize the fill value. If we add support for this, it'll be done globally through the `constants.py` module, so using the important constant would take those changes into account
2. I've thought about this and it does seem redundant, but for now we keep it since it's an expected attribute according to the UGRID conventions

Reply:
I can change back to the imported constant.

However, this Fill_Value design here might lead to a big loophole: Basically,
 we are exposing an API access that can modify the `_FillValue` attrs to users, while this `_FillValue` modification is not linked to the `constant.py` module.
 
 So users might do the following
 ```python
 
 # Use the default constant fill_value
 verts = [1.001, 1.001, INT_FILLVALUE]
 vgrid = Grid(verts, fillvalue = INT_FILLVALUE)
 
 # User thinks they are changing the fill value globally:
 vgrid.ds["Mesh2_face_nodes"].attrs["_FillValue"] = -1
 
 # And if they call our connectivity population function, the codes break down
 vgrid.populate_connectivity()
 ```
 
 This is also the reason why I am so conservative when incorporating the "FIll_VALUE" check in my functions. Never too much for a more adaptive/strict function call/check